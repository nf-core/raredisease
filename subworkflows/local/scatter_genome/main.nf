//
// A subworkflow to create genome interval files necessary for bam/vcf scatter operations.
//

include { BUILD_BED            } from '../../../modules/local/create_bed_from_fai'
include { GATK4_SPLITINTERVALS } from '../../../modules/nf-core/gatk4/splitintervals/main'

workflow SCATTER_GENOME {

    take:
        ch_genome_dictionary   // channel: [mandatory] [ val(meta), path(dict) ]
        ch_genome_fai    // channel: [mandatory] [ val(meta), path(fai) ]
        ch_genome_fasta  // channel: [mandatory] [ val(meta), path(fasta) ]

    main:
        ch_versions = Channel.empty()

        BUILD_BED (ch_genome_fai)

        GATK4_SPLITINTERVALS(BUILD_BED.out.bed, ch_genome_fasta, ch_genome_fai, ch_genome_dictionary)

        ch_versions = ch_versions.mix(BUILD_BED.out.versions)
        ch_versions = ch_versions.mix(GATK4_SPLITINTERVALS.out.versions)

    emit:
        bed             = BUILD_BED.out.bed.collect()   // channel: [ val(meta), path(bed) ]
        split_intervals = GATK4_SPLITINTERVALS.out.split_intervals.map { meta, it -> it }.flatten().collate(1) // channel: [ val(meta), [ path(interval_lists) ] ]
        versions        = ch_versions                   // channel: [ path(versions.yml) ]
}
