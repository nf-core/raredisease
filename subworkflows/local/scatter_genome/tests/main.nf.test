nextflow_workflow {

    name "Test Workflow SCATTER_GENOME"
    script "subworkflows/local/scatter_genome/main.nf"
    workflow "SCATTER_GENOME"
    config "./nextflow.config"

    test("Should run without failures") {

        when {
            params {
                raredisease_testdata_base_path= 'https://raw.githubusercontent.com/nf-core/test-datasets/raredisease/'
                fasta              = params.raredisease_testdata_base_path+ 'reference/reference.fasta'
                fai                = params.raredisease_testdata_base_path+ 'reference/reference.fasta.fai'
                dict               = params.raredisease_testdata_base_path+ 'reference/reference.dict'
                scatter_count      = 20
            }
            workflow {
                """
                input[0] = Channel.of(params.dict).map {it -> [[id:'genome'], it]}.collect()
                input[1] = Channel.of(params.fai).map {it -> [[id:'genome'], it]}.collect()
                input[2] = Channel.of(params.fasta).map {it -> [[id:'genome'], it]}.collect()
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
