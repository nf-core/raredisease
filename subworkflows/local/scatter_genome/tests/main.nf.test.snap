{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "genome"
                        },
                        "reference.fasta.bed:md5,77b5bd8a06d9f25356c233413bf0318d"
                    ]
                ],
                "1": [
                    [
                        "0000-scattered.interval_list:md5,91c71c99e1a0777d96ed2ed35977fe68"
                    ],
                    [
                        "0001-scattered.interval_list:md5,8084a75ace330aaa6b5baedcbc408e73"
                    ],
                    [
                        "0002-scattered.interval_list:md5,61a7e1852764653287dfcca76b177646"
                    ],
                    [
                        "0003-scattered.interval_list:md5,9f817fda35d09c8d449d7d91bc57d5a2"
                    ],
                    [
                        "0004-scattered.interval_list:md5,6f023cded659d4cfc8edb96706c3b319"
                    ],
                    [
                        "0005-scattered.interval_list:md5,91c3bca4311e2cc9bf55d4ef22281dd4"
                    ],
                    [
                        "0006-scattered.interval_list:md5,8f2a2f98ac6994859350580a8161cd91"
                    ],
                    [
                        "0007-scattered.interval_list:md5,5541b78eee72200b98be6af5e4544ed0"
                    ],
                    [
                        "0008-scattered.interval_list:md5,9f7f7b6f58ec47a73317e3fd64f53bac"
                    ],
                    [
                        "0009-scattered.interval_list:md5,0bd63474d5fb5284cb8bbb0a202f2e1e"
                    ],
                    [
                        "0010-scattered.interval_list:md5,c2ff40fe4d17059d38cf40b9b031cb88"
                    ],
                    [
                        "0011-scattered.interval_list:md5,dd86a813319aa10ca2694505799b8fbe"
                    ],
                    [
                        "0012-scattered.interval_list:md5,b9424bc405dac67b220e212a749844e0"
                    ],
                    [
                        "0013-scattered.interval_list:md5,60eee5b05fa8e3d4ff7f026071c4d0e4"
                    ],
                    [
                        "0014-scattered.interval_list:md5,8c35fb4f332a8bfe56a5d12269def003"
                    ],
                    [
                        "0015-scattered.interval_list:md5,3fff10ccee5d170b92804ed116f41760"
                    ],
                    [
                        "0016-scattered.interval_list:md5,c36e542cad7d5b534bde590ce26f356e"
                    ],
                    [
                        "0017-scattered.interval_list:md5,40c5de97da683709819f0c7404ea6411"
                    ],
                    [
                        "0018-scattered.interval_list:md5,81820b1a9b029868c58e1abc41c3dc03"
                    ],
                    [
                        "0019-scattered.interval_list:md5,c4e27e8be0b085a4fa4980d756e21c7f"
                    ]
                ],
                "2": [
                    "versions.yml:md5,005c9777d9befd3821945bc05709dfc8",
                    "versions.yml:md5,27694de4c876f5d1aeb42639252e73ea"
                ],
                "bed": [
                    [
                        {
                            "id": "genome"
                        },
                        "reference.fasta.bed:md5,77b5bd8a06d9f25356c233413bf0318d"
                    ]
                ],
                "split_intervals": [
                    [
                        "0000-scattered.interval_list:md5,91c71c99e1a0777d96ed2ed35977fe68"
                    ],
                    [
                        "0001-scattered.interval_list:md5,8084a75ace330aaa6b5baedcbc408e73"
                    ],
                    [
                        "0002-scattered.interval_list:md5,61a7e1852764653287dfcca76b177646"
                    ],
                    [
                        "0003-scattered.interval_list:md5,9f817fda35d09c8d449d7d91bc57d5a2"
                    ],
                    [
                        "0004-scattered.interval_list:md5,6f023cded659d4cfc8edb96706c3b319"
                    ],
                    [
                        "0005-scattered.interval_list:md5,91c3bca4311e2cc9bf55d4ef22281dd4"
                    ],
                    [
                        "0006-scattered.interval_list:md5,8f2a2f98ac6994859350580a8161cd91"
                    ],
                    [
                        "0007-scattered.interval_list:md5,5541b78eee72200b98be6af5e4544ed0"
                    ],
                    [
                        "0008-scattered.interval_list:md5,9f7f7b6f58ec47a73317e3fd64f53bac"
                    ],
                    [
                        "0009-scattered.interval_list:md5,0bd63474d5fb5284cb8bbb0a202f2e1e"
                    ],
                    [
                        "0010-scattered.interval_list:md5,c2ff40fe4d17059d38cf40b9b031cb88"
                    ],
                    [
                        "0011-scattered.interval_list:md5,dd86a813319aa10ca2694505799b8fbe"
                    ],
                    [
                        "0012-scattered.interval_list:md5,b9424bc405dac67b220e212a749844e0"
                    ],
                    [
                        "0013-scattered.interval_list:md5,60eee5b05fa8e3d4ff7f026071c4d0e4"
                    ],
                    [
                        "0014-scattered.interval_list:md5,8c35fb4f332a8bfe56a5d12269def003"
                    ],
                    [
                        "0015-scattered.interval_list:md5,3fff10ccee5d170b92804ed116f41760"
                    ],
                    [
                        "0016-scattered.interval_list:md5,c36e542cad7d5b534bde590ce26f356e"
                    ],
                    [
                        "0017-scattered.interval_list:md5,40c5de97da683709819f0c7404ea6411"
                    ],
                    [
                        "0018-scattered.interval_list:md5,81820b1a9b029868c58e1abc41c3dc03"
                    ],
                    [
                        "0019-scattered.interval_list:md5,c4e27e8be0b085a4fa4980d756e21c7f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,005c9777d9befd3821945bc05709dfc8",
                    "versions.yml:md5,27694de4c876f5d1aeb42639252e73ea"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-31T11:51:27.737317"
    }
}